src/pybind/mgr/orchestrator/module.py

```diff
@@ -369,11 +369,20 @@ def _update_set_addr(self, hostname: str, addr: str) -> HandleCommandResult:
         return HandleCommandResult(stdout=completion.result_str())
 
     @_cli_read_command('orch host ls')
-    def _get_hosts(self, format: Format = Format.plain) -> HandleCommandResult:
+    def _get_hosts(self, format: Format = Format.plain, matches_name: str = '', label: str = '', host_status: str = '') -> HandleCommandResult:
         """List hosts"""
         completion = self.get_hosts()
         hosts = raise_if_exception(completion)
 
+        if label:
+            hosts = [h for h in hosts if label in h.labels]
+
+        if matches_name:
+            hosts = [h for h in hosts if re.search(matches_name, h.hostname)]
```

Comment:
ok at least internally, I'd make use of

https://github.com/ceph/ceph/blob/602cbea5ed38f2120e5b31d35c24774331601448/src/python-common/ceph/deployment/service_spec.py#L196

to get consistency across different CLI calls. Otherwise we're introducing regular expressions just for this call here.